crates/findex/src/pyo3/py_callbacks.rs

```diff
@@ -0,0 +1,320 @@
+                            .map(|indexed_value| match indexed_value {
+                                IndexedValueRust::Location(location) => {
+                                    LocationPy(location.clone()).into_py(py)
+                                }
+                                IndexedValueRust::NextKeyword(keyword) => {
+                                    KeywordPy(keyword.clone()).into_py(py)
+                                }
+                            })
+                            .collect::<Vec<PyObject>>(),
+                    )
+                })
+                .collect::<HashMap<_, _>>();
+
+            let ret = self
+                .progress_callback
+                .call1(py, (py_results,))
+                .map_err(|e| FindexPyo3Error::Callback(format!("{e} (progress_callback)")))?;
+
+            ret.extract(py)
+                .map_err(|e| FindexPyo3Error::ConversionError(format!("{e} (progress_callback)")))
+        })
+    }
+
+    async fn fetch_all_entry_table_uids(
+        &self,
+    ) -> Result<HashSet<Uid<UID_LENGTH>>, FindexPyo3Error> {
+        Python::with_gil(|py| {
+            let results = self.fetch_all_entry_table_uids.call0(py).map_err(|e| {
+                FindexPyo3Error::Callback(format!("{e} (fetch_all_entry_table_uids)"))
+            })?;
+            let py_result_table: HashSet<[u8; UID_LENGTH]> = results
+                .extract(py)
+                .map_err(|e| FindexPyo3Error::ConversionError(format!("{e} (fetch_entry)")))?;
+
+            // Convert python result (HashSet<[u8; UID_LENGTH]>) to
+            // HashSet<Uid<UID_LENGTH>>
+            let entry_table_items = py_result_table
+                .into_iter()
+                .map(Uid::from)
+                .collect::<HashSet<_>>();
+            Ok(entry_table_items)
+        })
+    }
+
+    async fn fetch_entry_table(
+        &self,
+        entry_table_uids: &HashSet<Uid<UID_LENGTH>>,
+    ) -> Result<EncryptedTable<UID_LENGTH>, FindexPyo3Error> {
+        Python::with_gil(|py| {
+            let py_entry_uids = entry_table_uids
+                .iter()
+                .map(|uid| PyBytes::new(py, uid))
+                .collect::<Vec<_>>();
+            let results = self
+                .fetch_entry
+                .call1(py, (py_entry_uids,))
+                .map_err(|e| FindexPyo3Error::Callback(format!("{e} (fetch_entry)")))?;
+            let py_result_table: HashMap<[u8; UID_LENGTH], Vec<u8>> = results
+                .extract(py)
+                .map_err(|e| FindexPyo3Error::ConversionError(format!("{e} (fetch_entry)")))?;
+
+            // Convert python result (HashMap<[u8; UID_LENGTH], Vec<u8>>) to
+            // EncryptedEntryTable<UID_LENGTH>
+            let entry_table_items = py_result_table
+                .into_iter()
+                .map(|(k, v)| (Uid::from(k), v))
+                .collect::<HashMap<_, _>>();
+
+            Ok(entry_table_items.into())
```

Comment:
Style only (same everywhere):
```suggestion
            Ok(py_result_table
                .into_iter()
                .map(|(k, v)| (Uid::from(k), v))
                .collect())
```